auth/user_mgt.go

```diff
@@ -561,6 +582,261 @@ func (c *baseClient) getUser(ctx context.Context, query *userQuery) (*UserRecord
+	err := id.validate()
+	if err != nil {
+		return err
+	}
+	req.phoneNumber = append(req.phoneNumber, id.PhoneNumber)
+	return nil
+}
+
+// A ProviderIdentifier is used for looking up an account by federated provider.
+//
+// See GetUsers function.
+type ProviderIdentifier struct {
+	ProviderID  string
+	ProviderUID string
+}
+
+func (id ProviderIdentifier) String() string {
+	return fmt.Sprintf("ProviderIdentifier{%s, %s}", id.ProviderID, id.ProviderUID)
+}
+
+func (id ProviderIdentifier) validate() error {
+	return validateProvider(id.ProviderID, id.ProviderUID)
+}
+
+func (id ProviderIdentifier) matchesUserRecord(ur *UserRecord) (bool, error) {
+	err := id.validate()
+	if err != nil {
+		return false, err
+	}
+	for _, userInfo := range ur.ProviderUserInfo {
+		if id.ProviderID == userInfo.ProviderID && id.ProviderUID == userInfo.UID {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (id ProviderIdentifier) populateRequest(req *getAccountInfoRequest) error {
+	err := id.validate()
+	if err != nil {
+		return err
+	}
+	req.federatedUserID = append(
+		req.federatedUserID,
+		federatedUserIdentifier{providerID: id.ProviderID, rawID: id.ProviderUID})
+	return nil
+}
+
+// A GetUsersResult represents the result of the GetUsers() API.
+type GetUsersResult struct {
+	// Set of UserRecords, corresponding to the set of users that were requested.
+	// Only users that were found are listed here. The result set is unordered.
+	Users []*UserRecord
+
+	// Set of UserIdentifiers that were requested, but not found.
+	NotFound []UserIdentifier
+}
+
+type federatedUserIdentifier struct {
+	providerID string
+	rawID      string
+}
+
+type getAccountInfoRequest struct {
+	localID         []string
+	email           []string
+	phoneNumber     []string
+	federatedUserID []federatedUserIdentifier
+}
+
+func (req *getAccountInfoRequest) build() map[string]interface{} {
+	var builtFederatedUserID []map[string]interface{}
+	for i := range req.federatedUserID {
+		builtFederatedUserID = append(builtFederatedUserID, map[string]interface{}{
+			"providerId": req.federatedUserID[i].providerID,
+			"rawId":      req.federatedUserID[i].rawID,
+		})
+	}
+	return map[string]interface{}{
+		"localId":         req.localID,
+		"email":           req.email,
+		"phoneNumber":     req.phoneNumber,
+		"federatedUserId": builtFederatedUserID,
+	}
+}
+
+func isUserFound(id UserIdentifier, urs [](*UserRecord)) (bool, error) {
+	for i := range urs {
+		match, err := id.matchesUserRecord(urs[i])
+		if err != nil {
+			return false, err
+		} else if match {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (c *baseClient) GetUsers(
+	ctx context.Context, identifiers []UserIdentifier,
+) (*GetUsersResult, error) {
+	if len(identifiers) == 0 {
+		return &GetUsersResult{[](*UserRecord){}, [](UserIdentifier){}}, nil
+	} else if len(identifiers) > maxGetAccountsBatchSize {
+		return nil, internal.Errorf(
```

Comment:
We don't use error codes for developer errors. Just return a normal error. Use error codes for back-end errors only.

Reply:
This is something we've been working on. Lets discuss some other time. go/firebase-error-handling if you need some extra context in the meantime.